packages/malloy-vscode/src/server/lenses/lenses.ts

```diff
@@ -111,24 +111,27 @@ export function getMalloyLenses(document: TextDocument): CodeLens[] {
         }
       });
     } else if (symbol.type === "sql") {
+      // TODO crs-sql-statement Cannot access these by name currently
       lenses.push({
         range: symbol.range.toJSON(),
         command: {
           title: "Run",
-          command: "malloy.runSqlQuery",
-          arguments: [symbol.name],
+          command: "malloy.runUnnamedSQLBlock",
+          arguments: [currentUnnamedSQLBlockIndex],
```

Comment:
@mtoy-googly-moogly Since the `.name` field of a SQL Block is now no longer the same as what you type in `sql: xyz is ...`, I have to access the named SQL blocks by index. Can `.name` be the given name if it's present?

Reply:
Every sql block has "name: digest-hash"

In the sqlBlockList returned by the translator, every entry also has an "as:" field which will either be the name assigned by the code, or a manufactured unique name.

Reply:
I was following the model/ pattern, which is that the "name" of a thing is always "thing.as || thing.name". If you''d prefer for the name to be the name and the hash to be in a field called "digest", we could do that.

Reply:
I'll make the "as" undefined for anonymous blocks next time I am in that code.

Reply:
Think of the "name" as like "the actual table name" which the user also doesn't choose. If I have a table structedf, the "name" is the physical name of the table and the "as:" is what the user calls it.

Reply:
🤷 That's fine by me!